web/source/text/prediction/modelManager.ts

```diff
@@ -24,6 +24,29 @@ namespace com.keyman.text.prediction {
     path: string;
   }
 
+  class TranscriptionContext implements Context {
+    left: string;
+    right?: string;
+
+    startOfBuffer: boolean;
+    endOfBuffer: boolean;
+
+    constructor(mock: Mock, config: Configuration) {
+      this.left = mock.getTextBeforeCaret();
```

Comment:
Should we use references like "left of caret" / "right of caret"?
Just thinking ahead to rtl

Reply:
RTL conceptually works the same for the backing store. It's at presentation layer that it differs.

Reply:
I'm merely following long-existing conventions in the KMW code.  Touch-alias elements have been supporting `getTextBeforeCaret` in exactly this context, regardless of ltr vs rtl, since before I joined the project.